### DISCUSSION TITLE
Command palette: Re-execute last shell command

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#173] [Ideas - done & released]
Add a command palette command that can be used to repeat the last executed shell command.

- The command palette command shows what shell command was last executed (alias or the actual shell command, if no alias is defined). Text: _Shell commands: Re-execute: **Alias or shell command**_.
- Naturally supports custom hotkey assignation. No default hotkey will be added, though.
- Add a setting do decide what kind of shell commands this feature can re-execute:
  - Shell commands executed via the command palette / hotkeys. **Edit 2023-06-23:** No setting will be made at the first phase. At this stage, only shell commands that were executed using command palette or hotkeys, can be re-executed using this new command.
  - Shell commands executed via menu events.
  - Shell commands executed via other events.
 - Add a setting for disabling this command, so that the command palette is not cluttered with a surplus command if a user never needs this command.

## Use case

This is suited for a situation where there are lot of shell commands that are frequently used, and there's no practical possibility to assign hotkeys for all of them. If a user has assigned hotkeys to all of their shell commands, then this feature is maybe not so useful to them.

**Comment by FelipeRearden** [comment]
Spectacular idea @Taitava !!!!!

> Text: Shell commands: Re-execute: Alias or shell command.

Just one comment about the command name related to the `Execute` that is now customizable.

Just checking if `Re-execute` is replacing `Execute` or not.

Shell commands: Execute : Re-execute: Alias or shell command.

Or

Shell commands: Re-execute : Alias or shell command.

**Reply by Taitava** [reply]
> Shell commands: Re-execute : Alias or shell command.

☝️ This! 🙂 

Good that you mentioned the customisability of `Execute: `. The same customisability could be added to this feature, too.

**Comment by Taitava** [comment]
# This is now released in `0.20.0`.
